### PR TITLE
Disable notebook backups when file size is too large

### PR DESCRIPTION
**Pull request by amunger**
This is a safer version of https://github.com/microsoft/vscode/pull/210185, just disable backups entirely so we wont be in a weird state where the file on disk has outputs that the backed up file is missing

### CODE REVIEW

File: src/vs/workbench/contrib/notebook/test/browser/notebookEditorModel.test.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ import { CellKind, NotebookData, TransientOptions } from 'vs/workbench/contrib/n
 import { NotebookFileWorkingCopyModel } from 'vs/workbench/contrib/notebook/common/notebookEditorModel';
 import { INotebookSerializer, INotebookService, SimpleNotebookProviderInfo } from 'vs/workbench/contrib/notebook/common/notebookService';
 import { setupInstantiationService } from 'vs/workbench/contrib/notebook/test/browser/testNotebookEditor';
+import { SnapshotContext } from 'vs/workbench/services/workingCopy/common/fileWorkingCopy';
 
 suite('NotebookFileWorkingCopyModel', function () {
```
Comment (bpasero) [inline src/vs/workbench/contrib/notebook/test/browser/notebookEditorModel.test.ts:24]: Is it possible to have a test here for the throw on large backup? Maybe by configuring the threshold for the test to a low length for triggering it?

File: src/vs/workbench/contrib/notebook/browser/notebook.contribution.ts [hunk 2]
```diff
@@ -1094,5 +1094,10 @@ configurationRegistry.registerConfiguration({
 			type: 'boolean',
 			default: true
 		},
+		[NotebookSetting.outputBackupSizeLimit]: {
+			markdownDescription: nls.localize('notebook.backup.sizeLimit', "The limit of notebook output size where notebook files will no longer be backed up for hot reload. Use 0 for unlimited."),
+			type: 'number',
+			default: 10000
```
Comment (bpasero) [inline src/vs/workbench/contrib/notebook/browser/notebook.contribution.ts:1100]: Is this value based on some measurements? It seems quite low, for example Luna Paint picked 100kb.

**Review by bpasero — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/notebook/browser/notebook.contribution.ts (hunk 2 above)
Comment (amunger) [reply src/vs/workbench/contrib/notebook/browser/notebook.contribution.ts:1100]: this is in Kilobytes, so 10 mb (that used to be in the description, I will add it back in)

Comment (bpasero) [reply src/vs/workbench/contrib/notebook/browser/notebook.contribution.ts:1100]: Hmm, I wonder if we should be consistent here with other settings and pick `B` instead?

Comment (amunger) [reply src/vs/workbench/contrib/notebook/browser/notebook.contribution.ts:1100]: maybe MB should be used? workbench.editorLargeFileConfirmation also does

Comment (bpasero) [reply src/vs/workbench/contrib/notebook/browser/notebook.contribution.ts:1100]: Oh yeah true, I was wrong, even the local history setting uses KB:

image

**Review by bpasero — APPROVED** [review]

**Pull request merged**